source/Fleet.cpp

```diff
@@ -488,7 +493,12 @@ void Fleet::Place(const System &system, list<shared_ptr<Ship>> &ships, bool carr
 // Do the randomization to make a ship enter or be in the given system.
 const System *Fleet::Enter(const System &system, Ship &ship, const System *source)
 {
-	if(system.Links().empty() || (source && !system.Links().count(source)))
+	bool canEnter = (source != nullptr);
+	if(!canEnter && any_of(system.Links().begin(), system.Links().end(),
+			[ship](const System *link) noexcept -> bool { return ship.GetGovernment()->AllowJumpingTo(*link); }))
+		canEnter = true;
```

Comment:
You can merge this into a single `bool canEnter = ....` line. Also you're copying the ship since you're capturing it by value, capture it by reference instead: `[&ship]`